pom.xml

```diff
@@ -33,6 +33,39 @@
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>copy-dependencies</id>
+                        <phase>prepare-package</phase>
+                        <goals>
+                            <goal>copy-dependencies</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.build.directory}/lib</outputDirectory>
```

Comment:
why in 'lib' ?

Reply:
to place them in a repository dedicated to the jar of the dependencies

Reply:
I think it's far better to create one fat/uber jar with everything in it (so no dependency given that every thing is in the jar)

Reply:
Help please.... I managed to create the fat/uber but the module-info.class is still in a "original-name_of_jar.jar" and I don't know where to put the module-info as the plugin to create the fat/uber remove the file from it's jar. (And I need to specify the path of module-info.class when executing the jar to get the list of packages

Reply:
An uber jar does not support a module-info, the idea of a module-info is to resolve the dependencies and a uber jar as no dependency, so the best is to filter the module-info.class out of the uber jar.

Why do you want to get the list of packages ?

Reply:
Brian should correct me if I'm wrong but when the user gives the path for the .jar, he can ask a package in particular, we want the list of package to get the user's requested package.

For me, it's the why we want the list of packages, but I am going to ask Brian